openjdk/mmtkBarrierSetC2.cpp

```diff
@@ -67,12 +67,61 @@ void MMTkBarrierSetC2::expand_allocate(
   // The initial slow comparison is a size check, the comparison
   // we want to do is a BoolTest::gt
   bool always_slow = false;
+  // Check if we need initial_slow_test
   int tv = x->_igvn.find_int_con(initial_slow_test, -1);
   if (tv >= 0) {
     always_slow = (tv == 1);
     initial_slow_test = NULL;
+  }
+
+  // We will do MMTk size check.
+  // The implementation tries not to change the current control flow, and instead evaluate allocation size
+  // with max non-los bytes and combine the result with initial_slow_test. If the size check fails, initial_slow_test
+  // should be evaluated to true, and we jump to the slowpath.
+
+  // The max non-los bytes from MMTk
+  int max_non_los_bytes = (int)get_max_non_los_default_alloc_bytes();
+  // Check if allocation size is constant
+  int const_size = x->_igvn.find_int_con(size_in_bytes, -1);
```

Comment:
Looks like `size_in_bytes ` has long type (line 99 below). So use `find_int_con` instead of `find_long_con` here may make this optimization never triggered?

Reply:
Good point. I will fix this.